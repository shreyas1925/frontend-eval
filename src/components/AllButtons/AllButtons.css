.all--buttons-container {
  display: flex;
  font-family: Lato, "Helvetica Neue", Helvetica, Arial, sans-serif
}

.left--buttons {
  padding-left: 22%;
}

.right--buttons {
  padding-left: 30%;
}
.right--buttons .button--content {
  display: flex;
}

.mini-content {
  margin-left: 35%;
}

.button--content {
  display: flex;
  padding: 5px;
  margin: 5px;
}
.button--content p {
  padding: 8px;
  font-weight: bold;
  font-size: 14px;
}

input.search {
  padding: 5px;
  border-radius: 15px;
  outline: none;
  width: 190px;
}

/* .dropdown {
  position: relative;
  display: inline-block;
}

.dropdown--content {
  display: none;
  position: absolute;
  background-color: #f9f9f9;
  box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
  z-index: 1;
}
 .dropdown--content:hover {
  display: block;
} */

@media screen and (max-width:600px)
{
    .all--buttons-container{
        display: flex;
        flex-direction: column;
    }
}