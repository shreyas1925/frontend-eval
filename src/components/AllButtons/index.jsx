import React, { useState } from "react";
import "./AllButtons.css";
const AllButtons = ({ events, setAllEvents, filterdata, setFilterdata }) => {
  console.log("sa", events);

  const [query, setQuery] = useState("");
  const handlesearch = (event) => {
    const getSearch = event.target.value;
    if (getSearch.length > 0) {
      const searchdata = events.filter((item) =>
        item.name.toLowerCase().includes(getSearch)
      );
      //  console.log(searchdata);
      setAllEvents(searchdata);
    } else {
      setFilterdata(filterdata);
    }
    setQuery(getSearch);
  };

  const handleAll = () => {
    setAllEvents(events);
  };
  const handleRegisters = () => {
    const registered = events.filter((item) => item.isRegistered === true);
    setAllEvents(registered);
  };

  const handleBookmarks = () => {
    const bookmarked = events.filter((item) => item.isBookmarked === true);
    setAllEvents(bookmarked);
  };

  const handleSeats = () => {
    const seats = events.filter((item) => item.areSeatsAvailable === true);
    setAllEvents(seats);
  };

  return (
    <div className="all--buttons-container" data-testid='all-btns'>
      <div className="left--buttons">
        <div className="button--content">
          <p>
            <i className="fa fa-filter"></i>
          </p>
          <p>
            Filter <i className="fa fa-caret-up" aria-hidden="true"></i>
          </p>
        </div>
        {/* <div className="dropdown">
        <div className="dropdown--content"> */}
          <div className="button--content">
            <input type="radio" name="content" onClick={handleAll} />
            <p>ALL</p>
          </div>
          <div className="button--content">
            <input type="radio" name="content" onClick={handleRegisters} />
            <p>REGISTERED</p>
          </div>
        {/* </div>
        </div> */}
        
      </div>
      <div className="right--buttons">
        <div className="button--content">
          <input
            type="text"
            className="search"
            placeholder="Search"
            value={query}
            onChange={handlesearch}
          />
          <p>
            <i className="fa fa-search" aria-hidden="true"></i>
          </p>
        </div>
        <div className="mini-content">
          <div className="button--content">
            <p>BOOKMARKED</p>
            <input type="radio" name="content" onClick={handleBookmarks} />
          </div>
          <div className="button--content">
            <p>SEATS AVAILABLE</p>
            <input type="radio" name="content" onClick={handleSeats} />
          </div>
        </div>
      </div>
    </div>
  );
};

export default AllButtons;
