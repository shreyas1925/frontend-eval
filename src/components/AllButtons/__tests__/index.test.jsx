
import { render, screen } from "@testing-library/react";
import AllButtons from "..";
describe("AllButtons", () => {
    it("should render correctly and create a snapshot", () => {
        const {asFragment } = render(<AllButtons />);
        expect(asFragment()).toMatchSnapshot();
      });

    it("should render correctly", () => {
        render(<AllButtons />);
        const element = screen.getByTestId("all-btns");
        expect(element).toBeInTheDocument();
    });
});
