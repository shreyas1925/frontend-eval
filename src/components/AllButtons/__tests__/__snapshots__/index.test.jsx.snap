// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllButtons should render correctly and create a snapshot 1`] = `
<DocumentFragment>
  <div
    class="all--buttons-container"
    data-testid="all-btns"
  >
    <div
      class="left--buttons"
    >
      <div
        class="button--content"
      >
        <p>
          <i
            class="fa fa-filter"
          />
        </p>
        <p>
          Filter 
          <i
            aria-hidden="true"
            class="fa fa-caret-up"
          />
        </p>
      </div>
      <div
        class="button--content"
      >
        <input
          name="content"
          type="radio"
        />
        <p>
          ALL
        </p>
      </div>
      <div
        class="button--content"
      >
        <input
          name="content"
          type="radio"
        />
        <p>
          REGISTERED
        </p>
      </div>
    </div>
    <div
      class="right--buttons"
    >
      <div
        class="button--content"
      >
        <input
          class="search"
          placeholder="Search"
          type="text"
          value=""
        />
        <p>
          <i
            aria-hidden="true"
            class="fa fa-search"
          />
        </p>
      </div>
      <div
        class="mini-content"
      >
        <div
          class="button--content"
        >
          <p>
            BOOKMARKED
          </p>
          <input
            name="content"
            type="radio"
          />
        </div>
        <div
          class="button--content"
        >
          <p>
            SEATS AVAILABLE
          </p>
          <input
            name="content"
            type="radio"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
